packages/host/app/commands/create-ai-assistant-room.ts

```diff
@@ -42,15 +42,37 @@ export default class CreateAiAssistantRoomCommand extends HostBaseCommand<
         'Requires userId to execute CreateAiAssistantRoomCommand',
       );
     }
-    let { defaultSkills } = input;
-    let skillFileDefs: FileDef[] | undefined;
+    let { enabledSkills, disabledSkills } = input;
+    let enabledSkillFileDefs: FileDef[] | undefined;
     let commandFileDefs: FileDef[] | undefined;
-    if (defaultSkills) {
-      let skills = input.defaultSkills;
-      skillFileDefs = await matrixService.uploadCards(input.defaultSkills);
-      const commandDefinitions = skills.flatMap((skill) => skill.commands);
-      commandFileDefs =
-        await matrixService.uploadCommandDefinitions(commandDefinitions);
+    let disabledSkillFileDefs: FileDef[] | undefined;
+
+    if (enabledSkills?.length) {
+      enabledSkillFileDefs = await matrixService.uploadCards(enabledSkills);
+    }
+    if (disabledSkills?.length) {
+      disabledSkillFileDefs = await matrixService.uploadCards(disabledSkills);
+    } else {
+      disabledSkillFileDefs = [];
+    }
+
+    // Combine command definitions from both enabled and disabled skills
+    const allCommandDefinitions = [
+      ...(enabledSkills?.flatMap((skill) => skill.commands) || []),
+      ...(disabledSkills?.flatMap((skill) => skill.commands) || []),
+    ];
+
+    // Remove duplicates based on command name or ID
+    const uniqueCommandDefinitions = allCommandDefinitions.filter(
+      (command, index, array) =>
+        array.findIndex((cmd) => command.functionName === cmd.functionName) ===
+        index,
+    );
+
+    if (uniqueCommandDefinitions.length) {
+      commandFileDefs = await matrixService.uploadCommandDefinitions(
+        uniqueCommandDefinitions,
+      );
```

Comment:
Doesn't this code exist somewhere already? Can we have it in one place instead of duplicating it here?